### ISSUE TITLE
Removing the product from the cart after applying the coupon gives "Sorry, it seems the coupon "xxxxxx" is invalid - it has now been removed from your order." error

### ISSUE THREAD

**Issue by chaitanyamsv** [woocommerce/woocommerce#25057]
**Describe the bug**

Removing the product from the cart after applying the coupon gives "Sorry, it seems the coupon "xxxxxx" is invalid - it has now been removed from your order." error

**To Reproduce**
1. During the check out apply a coupon.
2. Remove the product from the cart by clicking the x icon in the checkout table.
3. "Sorry, it seems the coupon "xxxxxx" is invalid - it has now been removed from your order." error appears.

**Screenshots**
Here is the screencast of the problem: https://cld.wthms.co/tyWpNp

**Expected behavior**
The product should be removed without showing any error with the coupon.

**Isolating the problem (mark completed items with an [x]):**
- [] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://kivitreats.shop
Site address (URL): https://kivitreats.shop
WC Version: 3.8.0
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.2.24
PHP Post Max Size: 2 GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.66.0
OpenSSL/1.1.0l

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.17-MariaDB-log
Max Upload Size: 2 GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.8.0
WC Database Prefix: wp_130622412_
MaxMind GeoIP Database: ✔
Total Database Size: 4.43MB
Database Data Size: 2.69MB
Database Index Size: 1.74MB
wp_130622412_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_130622412_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.06MB + Engine InnoDB
wp_130622412_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_130622412_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_130622412_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_130622412_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_comments: Data: 0.25MB + Index: 0.38MB + Engine InnoDB
wp_130622412_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_options: Data: 1.28MB + Index: 0.06MB + Engine InnoDB
wp_130622412_postmeta: Data: 0.16MB + Index: 0.11MB + Engine InnoDB
wp_130622412_posts: Data: 0.23MB + Index: 0.16MB + Engine InnoDB
wp_130622412_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_usermeta: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_130622412_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_130622412_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_130622412_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_130622412_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_130622412_wc_order_stats: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_130622412_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_130622412_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_130622412_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_130622412_woocommerce_shipping_table_rates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (20) ###

Akismet Anti-Spam: by Automattic – 4.1.3
AMP: by AMP Project Contributors – 1.4.1
Change Quantity on Checkout for WooCommerce: by Bhavik Kiri – 1.0 – Not tested with the active version of WooCommerce
Homepage Control: by WooThemes – 2.0.3
Jetpack by WordPress.com: by Automattic – 7.9
Storefront Blog Customiser: by WooCommerce – 1.3.0
Storefront Footer Bar: by WooThemes – 1.0.4
Storefront Homepage Contact Section: by WooThemes – 1.0.5
Storefront Parallax Hero: by WooCommerce – 1.5.7
Storefront Powerpack: by WooCommerce – 1.4.15 – 1.5.0 is available
Storefront Reviews: by WooCommerce – 1.0.5 – 1.0.6 is available – Not tested with the active version of WooCommerce
Storefront Sticky Add to Cart: by WooThemes – 1.1.9
WooCommerce Admin: by WooCommerce – 0.22.0
WooCommerce Empty Cart Button: by Technorigins – 1.1 – Not tested with the active version of WooCommerce
Google Ads for WooCommerce: by Kliken – 1.0.1 – Not tested with the active version of WooCommerce
WooCommerce Google Analytics Integration: by WooCommerce – 1.4.15
WooCommerce Min/Max Quantities: by WooCommerce – 2.4.6 – 2.4.12 is available – Not tested with the active version of WooCommerce
WooCommerce Product Reviews Pro: by SkyVerge – 1.13.3 – 1.15.0 is available – Not tested with the active version of WooCommerce
WooCommerce Table Rate Shipping: by WooCommerce – 3.0.15 – 3.0.18 is available – Not tested with the active version of WooCommerce
WooCommerce: by Automattic – 3.8.0

### Inactive Plugins (4) ###

Classic Editor: by WordPress Contributors – 1.5
Full Site Editing: by Automattic – 0.14
Gutenberg: by Gutenberg Team – 6.9.0
WooCommerce Helper: by WooCommerce – 1.7.2 – Not tested with the active version of WooCommerce – Network enabled

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: by  –

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: INR (₹)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #41 - /shop/
Cart: #42 - /cart/
Checkout: #43 - /checkout/
My account: #44 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Toyshop
Version: 2.0.19
Author URL: https://woocommerce.com
Child Theme: ✔
Parent Theme Name: Storefront
Parent Theme Version: 2.2.4 – 2.5.3 is available
Parent Theme Author URL: https://woocommerce.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 515
Oldest: 2019-10-18 01:49:14 +0000
Newest: 2019-11-18 01:30:01 +0000

Pending: 2
Oldest: 2019-11-18 02:30:01 +0000
Newest: 2019-11-21 05:34:09 +0000

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

`
```
</details>

**Comment by juliaamosova** [comment]
Thank you opening this issue @chaitanyamsv. However, I can't reproduce it. 

See my screencast below - I am not getting an error message when removing the product from the cart after applying coupon:

http://g.recordit.co/z6HtAINGTY.gif

Your screencast is made on the site that runs a non-default WooCommerce theme (such as Storefront or Twenty Nineteen). Can you please confirm that you can reproduce this error with only WooCommerce and default theme activated? I see that you marked it as being done on your bug report, however, I always like to double-check.

**Comment by nagabrahmam-mantha** [comment]
@chaitanyamsv Most likely the coupon code is restricted to be used only on specific products. So when you remove those specific products from the cart the coupon code is automatically removed. I have such products/coupons on my site. Hope that helps.

**Comment by chaitanyamsv** [comment]
@juliaamosova I spent some time testing again on my original test store that's on [Toyshop](https://woocommerce.com/products/toyshop/) (Storefront child theme), and I couldn't reproduce it anymore!

Although I did not restrict the coupon to a specific product,  I suspect I had **Minimum spend** set while testing. I likely noticed the coupon error when the cart amount went below the minimum spend as I removed some products from the cart.

I also tested again in the default Storefront theme as well as Independent Publisher theme and I couldn't reproduce this issue. Sorry for the false report. I think we can close this now.

Thanks for chiming in, @nagabrahmam-mantha!
